crates/goose/src/providers/formats/openai.rs

```diff
@@ -267,7 +276,9 @@ pub fn create_request(
         vec![]
     };
 
-    let mut messages_array = vec![system_message];
+    if !is_o1_mini {
+        messages_array.push(system_message);
```

Comment:
this is trickier. if we keep o1-mini, we would probably wanna prepend system msg content to the first user msg. we can skip this since we'd reocmmend gpt-4o over o1-mini anyway